doc/design/traingjob_crd_design.md

```diff
@@ -0,0 +1,172 @@
+ type trainingJobEventType string
+ 
+ const (
+ 	trainingJobEventDelete trainingJobEventType = "Delete"
+ 	trainingJobEventModify trainingJobEventType = "Modify"
+ )
+ 
+ type trainingJobEvent struct {
+ 	// pet is the TrainingJobEventType of TrainingJob
+ 	pet trainingJobEventType
+ 	// The job transfer the information fo job
+ 	job *v1alpha1.TrainingJob
+ }
+ 
+ // TrainingJober is to manager a specific TrainingJob
+ type TrainingJober struct {
+ 	// job is the job the TrainingJober manager.
+ 	job *v1alpha1.TrainingJob
+ 
+ 	// kubeCli is standard kubernetes client.
+ 	kubeCli kubernetes.Interface
+ 
+ 	// trainingJobClient is the client of TrainingJob.
+ 	trainingJobClient trainingJobClient.Interface
+ 
+ 	// status is the status in memory, update when TrainingJob status changed and update the CRD resource status.
+ 	status v1alpha1.TrainingJobStatus
+ 
+ 	// eventCh is the channel received by Controller, include Modify and Delete.
+ 	// When trainingJobEvent is Delete it will delete all resources
+ 	// The maximum is 1000.
+ 	eventCh chan *trainingJobEvent
+ }
+ 
+ ```
+ When user submit a TrainingJob, controller start a TrainingJober to manage the TrainingJob. 
+ - It will parser the config to TrainingJob spec including PSERVER, MASTER and TRAINER.
+ - It will create resource orderly. 
+ - It will sync the status of the job at regular time. While the status changed, it will update the job's status to k8s. 
+ - It will release all the resource while the job succeeded or failed.
+ 
+### Controller
+
+The controller manages distributed PaddlePaddle jobs by creating a series of TrainingJobers. Here is the struct of 
+Controller.
+```go
+type Controller struct {
+	// KubeCli is a standard kubernetes clientset
+	KubeCli kubernetes.Interface
+	// ApiCli is the extension kubernetes clientset
+	ApiCli apiextensionsclient.Interface
+	// PaddleCli is a clientset for our own API group
+	PaddleCli paddleclientset.Interface
+
+	trainingjobLister paddlelisters.TrainingJobLister
+	trainingjobSynced cache.InformerSynced
+
+	jobtracker map[string]*trainingjober.TrainingJober
+
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens.
+	workqueue workqueue.RateLimitingInterface
+	// recorder is an event recorder for recording Event resources to the
+	// Kubernetes API.
+	recorder record.EventRecorder
+}
+
+```
+
+- It start to confirm that there are only one TrainingJob controller in cluster.
+- It will register TrainingJob CRD to cluster if there is no TrainingJob resource.
+- It will create a TrainingJober while PaddlePaddle job was submitted and delete the TrainingJober while job was 
+deleted.
+
+### Overall design
+
+The whole life cycle of TrainingJob is managed through the two layer control of Controller and TrainingJober. As 
+shown in the following figure:
+
+![image](https://github.com/qizheng09/figure/blob/master/paddlecloud/overall_design.png?raw=true)
```

Comment:
Put the image in this repo. And, if you are using tools like "OmniGraffle" or something like that, you should put the source file in the repo too, so we can modify the graph later for updates.

Reply:
This figure should have some explanation of how it runs and what components it consists of.